test/test_product_config.py

```diff
@@ -1710,55 +1717,55 @@ async def test_sim(self) -> None:
         assert streams[2].src_streams[0] is config.streams[1]
         assert streams[1].src_streams[0] is config.streams[0]
 
-    async def test_cyclic(self) -> None:
-        self.config_sim['outputs']['bcp']['src_streams'] = ['vis']
+    async def test_cyclic(self, config_sim: Dict[str, Any]) -> None:
+        config_sim['outputs']['bcp']['src_streams'] = ['vis']
         with pytest.raises(ValueError):
-            await Configuration.from_config(self.config_sim)
+            await Configuration.from_config(config_sim)
 
-    async def test_bad_stream(self) -> None:
-        self.config['outputs']['l0']['continuum_factor'] = 3
+    async def test_bad_stream(self, config: Dict[str, Any]) -> None:
```

Comment:
How do you know if it picks `TestConfiguration.config` or the module-level `config`? Pytest magic?

Reply:
pytest has similar scoping rules to Python. Class takes precedence over the module (and the module over conftest.py).